code/modules/atmospherics/machinery/components/machines/atmos_machines_recipes.dm

```diff
@@ -0,0 +1,181 @@
+	for(var/path in meta_recipe)
+		if(meta_recipe[path][META_RECIPE_ID] == id)
+			return path
+	return ""
+
+/datum/gas_recipe
+	///Id of the recipe for easy identification in the code
+	var/id = ""
+	///What machine the recipe is for
+	var/machine_type = ""
+	///Displayed name of the recipe
+	var/name = ""
+	///Minimum temperature for the recipe
+	var/min_temp = TCMB
+	///Maximum temperature for the recipe
+	var/max_temp = INFINITY
+	///Type of reaction (either endothermic or exothermic)
+	var/reaction_type = ""
+	///Amount of energy released/consumed by the reaction (always positive)
+	var/energy_release = 0
+	///Gas required for the recipe to work
+	var/list/requirements
+	///Products made from the machine
+	var/list/products
+
+/datum/gas_recipe/crystallizer
+	machine_type = "Crystallizer"
+
+/datum/gas_recipe/crystallizer/metallic_hydrogen
+	id = "metal_h"
+	name = "Metallic hydrogen"
+	min_temp = 50000
+	max_temp = 150000
+	reaction_type = "ENDOTHERMIC_REACTION"
+	energy_release = 250000
+	requirements = list(/datum/gas/hydrogen = 600, /datum/gas/bz = 200)
+	products = list(/obj/item/stack/sheet/mineral/metal_hydrogen = 2)
+
+/datum/gas_recipe/crystallizer/healium_grenade
+	id = "healium_g"
+	name = "Healium crystal"
+	min_temp = 200
+	max_temp = 400
+	reaction_type = "ENDOTHERMIC_REACTION"
+	energy_release = 100000
+	requirements = list(/datum/gas/healium = 400, /datum/gas/freon = 800, /datum/gas/plasma = 50)
+	products = list(/obj/item/grenade/gas_crystal/healium_crystal = 1)
+
+/datum/gas_recipe/crystallizer/proto_nitrate_grenade
+	id = "proto_nitrate_g"
+	name = "Proto nitrate crystal"
+	min_temp = 200
+	max_temp = 400
+	reaction_type = "EXOTHERMIC_REACTION"
+	energy_release = 150000
+	requirements = list(/datum/gas/proto_nitrate = 400, /datum/gas/nitrogen = 800, /datum/gas/oxygen = 800)
+	products = list(/obj/item/grenade/gas_crystal/proto_nitrate_crystal = 1)
+
+/datum/gas_recipe/crystallizer/hot_ice
+	id = "hot_ice"
+	name = "Hot ice"
+	min_temp = 15
+	max_temp = 35
+	reaction_type = "ENDOTHERMIC_REACTION"
+	energy_release = 300000
+	requirements = list(/datum/gas/freon = 500, /datum/gas/plasma = 400, /datum/gas/oxygen = 300)
+	products = list(/obj/item/stack/sheet/hot_ice = 3)
+
+/datum/gas_recipe/crystallizer/ammonia_crystal
+	id = "ammonia_crystal"
+	name = "Ammonia crystal"
+	min_temp = 200
+	max_temp = 240
+	reaction_type = "EXOTHERMIC_REACTION"
+	energy_release = 15000
+	requirements = list(/datum/gas/hydrogen = 500, /datum/gas/nitrogen = 400)
+	products = list(/obj/item/stack/ammonia_crystals = 4)
+
+/datum/gas_recipe/crystallizer/shard
+	id = "crystal_shard"
+	name = "Supermatter crystal shard"
+	min_temp = TCMB
+	max_temp = 5
+	reaction_type = "EXOTHERMIC_REACTION"
+	energy_release = 1500000
+	requirements = list(/datum/gas/hypernoblium = 1500, /datum/gas/antinoblium = 1500, /datum/gas/bz = 2000, /datum/gas/plasma = 5000, /datum/gas/oxygen = 4500)
+	products = list(/obj/machinery/power/supermatter_crystal/shard = 1)
```

Comment:
i think it would be an interesting take for atmosians to create something like a SM shard, yes you can buy them, but these will come with a purity modifier on the name. the cost can be lowered if deemed to high
i think we'll be able to see these made

Reply:
How long do you think it'll take to make them?

Reply:
one and half hour with 2/3 atmosians distributing the workload
a couple if you are alone and know how to do this.
the hard part is the antinoblium gas only